### DISCUSSION TITLE
Expanded Reference Patterns with Regex

### DISCUSSION THREAD

**Discussion by zoopsel** [inventree/InvenTree#4479] [Q&A]
Hi Inventree Team

We use the reference patterns for BO/PO in a following way: `BO-{ref:06d}-??-###`
Where the two questionmarks are initials. However, now there is the problem that some initials require three letters. With the tools given it is not possible to use two or more characters for the initials.

One solution that I tried, would be to comment out the escaping of regex characters in the InvenTree/format.py line 80/81.
This way we could use the pattern `BO-{ref:06d}-[A-Z]+-###`, which would allow two or more characters as initials in the reference pattern.

I wanted to ask if that might break something behind the scenes that I can't spot immediately with some testing on the website?

**Comment by SchrodingersGat** [comment]
Not "immediately" ;) You would have to test it and see. It has been a while since I implemented that particular part of the code, but I imagine it should be ok to try what you are saying. I'd be happy to look at a PR to allow for more complex regex sequences here
